### ISSUE TITLE
Uniformity checking: issues with UniGen 

### ISSUE THREAD

**Issue by acherm** [FAMILIAR-project/usampling-exp#13]
`Namespace(delta=0.05, epsilon=0.3, eta=0.9, exp=1, formulas=['/home/samplingfm/Benchmarks/V15/s420_15_7.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring24.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring12.cnf', '/home/samplingfm/Benchmarks/V7/s953a_7_4.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring1.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring4.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case47.cnf', '/home/samplingfm/Benchmarks/V7/s526a_7_4.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring9.cnf', '/home/samplingfm/Benchmarks/V3/s349_3_2.cnf', '/home/samplingfm/Benchmarks/V15/s510_15_7.cnf', '/home/samplingfm/Benchmarks/V3/s420_new1_3_2.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_0_b11_1.cnf', '/home/samplingfm/Benchmarks/V15/s820a_15_7.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring6.cnf', '/home/samplingfm/Benchmarks/V3/s953a_3_2.cnf', '/home/samplingfm/Benchmarks/V15/s953a_15_7.cnf', '/home/samplingfm/Benchmarks/V15/s526_15_7.cnf', '/home/samplingfm/Benchmarks/V3/s420_new_3_2.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case35.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring26.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring2.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case114.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case55.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case34.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case205.cnf', '/home/samplingfm/Benchmarks/V7/s344_7_4.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case109.cnf', '/home/samplingfm/Benchmarks/V3/s526a_3_2.cnf', '/home/samplingfm/Benchmarks/V15/s444_15_7.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring20.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case116.cnf', '/home/samplingfm/Benchmarks/V7/s420_new1_7_4.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case5.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case9.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_1_b12_2.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case8.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring28.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_1_b11_1.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_3_b14_1.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case123.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case106.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case57.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_3_b14_3.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case201.cnf', '/home/samplingfm/Benchmarks/V3/s1238a_3_2.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring27.cnf', '/home/samplingfm/Benchmarks/V3/s510_3_2.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case126.cnf', '/home/samplingfm/Benchmarks/V7/s420_new_7_4.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_2_b12_2.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring21.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_0_b12_1.cnf', '/home/samplingfm/Benchmarks/V7/s820a_7_4.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring51.cnf', '/home/samplingfm/Benchmarks/V7/s382_7_4.cnf', '/home/samplingfm/Benchmarks/V3/s344_3_2.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case133.cnf', '/home/samplingfm/Benchmarks/V7/s510_7_4.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case101.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case120.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring22.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case2.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_2_b12_1.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring14.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case202.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring5.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case14.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_1_b14_1.cnf', '/home/samplingfm/Benchmarks/V15/s420_new_15_7.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case203.cnf', '/home/samplingfm/Benchmarks/V7/s444_7_4.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case3.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring10.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case135.cnf', '/home/samplingfm/Benchmarks/V7/s526_7_4.cnf', '/home/samplingfm/Benchmarks/V3/s526_3_2.cnf', '/home/samplingfm/Benchmarks/V15/s349_15_7.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case54.cnf', '/home/samplingfm/Benchmarks/V15/s1238a_15_7.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring30.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring25.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring7.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_2_b14_3.cnf', '/home/samplingfm/Benchmarks/V15/s1196a_15_7.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_1_b12_1.cnf', '/home/samplingfm/Benchmarks/V15/s344_15_7.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_2_b14_2.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case105.cnf', '/home/samplingfm/Benchmarks/V7/s420_7_4.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case122.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring16.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case61.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_0_b12_2.cnf', '/home/samplingfm/Benchmarks/V15/s420_new1_15_7.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case111.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case143.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case204.cnf', '/home/samplingfm/Benchmarks/V3/s420_3_2.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring50.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring3.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case115.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case132.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring8.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring11.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_1_b14_2.cnf', '/home/samplingfm/Benchmarks/V3/s382_3_2.cnf', '/home/samplingfm/Benchmarks/V3/s444_3_2.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case145.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case131.cnf', '/home/samplingfm/Benchmarks/V7/s1238a_7_4.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring23.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case121.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case56.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case124.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_3_b14_2.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case119.cnf', '/home/samplingfm/Benchmarks/V3/s1196a_3_2.cnf', '/home/samplingfm/Benchmarks/V7/s832a_7_4.cnf', '/home/samplingfm/Benchmarks/V7/s349_7_4.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case146.cnf', '/home/samplingfm/Benchmarks/V7/s1196a_7_4.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case62.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_1_b14_3.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case6.cnf', '/home/samplingfm/Benchmarks/V15/s832a_15_7.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring29.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case100.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case136.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case_2_b14_1.cnf', '/home/samplingfm/Benchmarks/V15/s382_15_7.cnf', '/home/samplingfm/Benchmarks/Blasted_Real/blasted_case125.cnf', '/home/samplingfm/Benchmarks/V15/s526a_15_7.cnf'], maxSamples=9223372036854775807, minSamples=0, sampler=1, searchOrder=0, seed=1, thres=43000, verbose=1)`

result: Uniform-UniGen.csv

```
file,time,cmd_output,err_output,Uniform,Timeout
/home/samplingfm/Benchmarks/V15/s420_15_7.cnf,4882.376,b'Experiment: 0     of     1\nconstantFactor:39   boundFactor: 14.541336623567457   logBoundFactor: 3.8620879811712894  \ntj: 1789   totalLoops: 2     beta: 0.8181818181818182 epsilon: 0.3       \nnumSolutions: 568   loThresh:82     hiThresh: 485   \n*** Progress *** i: 0 j: 2 ---- 0 / 3578\nUsing sampler: UniGen\ncmd:  ./samplers/unigen --samples=1 /home/samplingfm/Benchmarks/V15/s420_15_7.cnf /tmp/s420_15_7.txt > /dev/null 2>&1\nNumber of solutions returned by sampler: 0\nUsing sampler: KUS\ncmd:  python3  KUS.py --samples=1 --outputfile /tmp/s420_15_7.txt /home/samplingfm/Benchmarks/V15/s420_15_7.cnf > /dev/null 2>&1\nNumber of solutions returned by sampler: 1\n',"b'Traceback (most recent call last):\n  File ""barbarik.py"", line 1089, in <module>\n    ok, breakExperiment = exp.one_experiment(experiment, j, i, numExperiments, tj)\n  File ""barbarik.py"", line 976, in one_experiment\n    chainFormulaConf = chainFormulaSetup(sampleSol, unifSol, self.numSolutions)\n  File ""barbarik.py"", line 664, in chainFormulaSetup\n    sampleSol = sampleSol[0].strip()\nIndexError: list index out of range\n'",N/A,FALSE
/home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring24.cnf,2510.333,b'Experiment: 0     of     1\nconstantFactor:39   boundFactor: 14.541336623567457   logBoundFactor: 3.8620879811712894  \ntj: 1789   totalLoops: 2     beta: 0.8181818181818182 epsilon: 0.3       \nnumSolutions: 568   loThresh:82     hiThresh: 485   \n*** Progress *** i: 0 j: 2 ---- 0 / 3578\nUsing sampler: UniGen\ncmd:  ./samplers/unigen --samples=1 /home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring24.cnf /tmp/blasted_squaring24.txt > /dev/null 2>&1\nNumber of solutions returned by sampler: 1\nUsing sampler: KUS\ncmd:  python3  KUS.py --samples=1 --outputfile /tmp/blasted_squaring24.txt /home/samplingfm/Benchmarks/Blasted_Real/blasted_squaring24.cnf > /dev/null 2>&1\n',"b'Traceback (most recent call last):\n  File ""barbarik.py"", line 1089, in <module>\n    ok, breakExperiment = exp.one_experiment(experiment, j, i, numExperiments, tj)\n  File ""barbarik.py"", line 973, in one_experiment\n    self.inputFile, 1, SAMPLER_KUS, self.indVarList, newSeed)\n  File ""barbarik.py"", line 196, in getSolutionFromSampler\n    sols = SolutionRetriver.getSolutionFromKUS(*topass_withseed)\n  File ""barbarik.py"", line 299, in getSolutionFromKUS\n    with open(tempOutputFile, \'r\') as f:\nFileNotFoundError: [Errno 2] No such file or directory: \'/tmp/blasted_squaring24.txt\'\n'",N/A,FALSE
```

basically there are two issues: a file not found in the temp directory... and another related to index bound of a list
